riff-raff/app/deployment/Deployments.scala

```diff
@@ -21,25 +22,29 @@ import scala.util.control.NonFatal
 class Deployments(deploymentEngine: DeploymentEngine) extends Logging {
   import Deployments._
 
-  def deploy(requestedParams: DeployParameters): UUID = {
+  def deploy(requestedParams: DeployParameters, requestSource: RequestSource): Either[String, UUID] = {
     log.info(s"Started deploying $requestedParams")
-    if (enableQueueingSwitch.isSwitchedOff)
-      throw new IllegalStateException("Unable to queue a new deploy; deploys are currently disabled by the %s switch" format enableQueueingSwitch.name)
-
-    val params = if (requestedParams.build.id != "lastSuccessful")
-      requestedParams
+    val restrictionsPreventingDeploy = RestrictionChecker.configsThatPreventDeployment(RestrictionConfigDynamoRepository,
+      requestedParams.build.projectName, requestedParams.stage.name, requestSource)
+    if (restrictionsPreventingDeploy.nonEmpty) {
+      Left(s"Unable to queue deploy as restrictions are currently in place: ${restrictionsPreventingDeploy.map(r => s"${r.fullName}: ${r.note}").mkString("; ")}")
```

Comment:
It was easy to do so I've created a small `case class Error(message: String) extends AnyVal`.